builtin/providers/azurerm/resource_arm_loadbalancer.go

```diff
@@ -0,0 +1,295 @@
+		if feIpcErr != nil {
+			return fmt.Errorf("Error Building list of Loadbalancer Frontend IP Configurations: %s", feIpcErr)
+		}
+		properties.FrontendIPConfigurations = &frontEndConfigs
+	}
+
+	loadbalancer := network.LoadBalancer{
+		Name:       azure.String(name),
+		Location:   azure.String(location),
+		Tags:       expandedTags,
+		Properties: &properties,
+	}
+
+	_, err := loadBalancerClient.CreateOrUpdate(resGroup, name, loadbalancer, make(chan struct{}))
+	if err != nil {
+		return err
+	}
+
+	read, err := loadBalancerClient.Get(resGroup, name, "")
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Loadbalancer %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	log.Printf("[DEBUG] Waiting for LoadBalancer (%s) to become available", name)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"Accepted", "Updating"},
+		Target:  []string{"Succeeded"},
+		Refresh: loadbalancerStateRefreshFunc(client, resGroup, name),
+		Timeout: 10 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Loadbalancer (%s) to become available: %s", name, err)
+	}
+
+	return resourceArmLoadbalancerRead(d, meta)
+}
+
+func resourceArmLoadbalancerRead(d *schema.ResourceData, meta interface{}) error {
+	loadBalancer, exists, err := retrieveLoadbalancerById(d.Id(), meta)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		d.SetId("")
+		log.Printf("[INFO] Loadbalancer %q not found. Refreshing from state", d.Get("name").(string))
+		return nil
+	}
+
+	if loadBalancer.Properties != nil && loadBalancer.Properties.FrontendIPConfigurations != nil {
+		d.Set("frontend_ip_configuration", flattenLoadBalancerFrontendIpConfiguration(loadBalancer.Properties.FrontendIPConfigurations))
+	}
+
+	flattenAndSetTags(d, loadBalancer.Tags)
+
+	return nil
+}
+
+func resourceArmLoadbalancerDelete(d *schema.ResourceData, meta interface{}) error {
+	loadBalancerClient := meta.(*ArmClient).loadBalancerClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["loadBalancers"]
+
+	_, err = loadBalancerClient.Delete(resGroup, name, make(chan struct{}))
+
+	return err
+}
+
+func resourceArmLoadbalancerFrontEndIpConfigurationHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
+
+	if m["private_ip_address"] != nil {
+		buf.WriteString(fmt.Sprintf("%s-", m["private_ip_address"].(string)))
+	}
+	if m["public_ip_address_id"] != nil {
+		buf.WriteString(fmt.Sprintf("%s-", m["public_ip_address_id"].(string)))
+	}
+	if m["subnet_id"] != nil {
+		buf.WriteString(fmt.Sprintf("%s-", m["subnet_id"].(string)))
+	}
+
+	return hashcode.String(buf.String())
+}
+
+func expandAzureRmLoadbalancerFrontendIpConfigurations(d *schema.ResourceData) ([]network.FrontendIPConfiguration, error) {
```

Comment:
Three is never a condition where we return an error here - are there error conditions or can we remove the error from the signature?